packages/fast-components-react-base/src/media/Media.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "../foundation";
+import { IMediaHandledProps, IMediaMangedClasses, IMediaSourceSet, IMediaUnhandledProps } from "./media.props";
+import { IManagedClasses, IMediaClassNameContract } from "@microsoft/fast-components-class-name-contracts";
+
+/* tslint:disable-next-line */
+class Media extends Foundation<IMediaHandledProps & IManagedClasses<IMediaClassNameContract>,  React.HTMLAttributes<HTMLImageElement | HTMLPictureElement>, {}> {
+    protected handledProps: HandledProps<IMediaHandledProps & IManagedClasses<IMediaClassNameContract>> = {
+        managedClasses: void 0
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLImageElement | HTMLPictureElement> {
+        return (
+             (this.props.src) ? this.generateSimpleImage() : this.generatePicture()
+        );
+    }
+
+    protected generateClassNames(): string {
```

Comment:
Suggestion - We could probably combine these two and just pass generateClassNames for each of these by passing a boolean value in the nested instance. Comments inline for clarity:

```
generateClassNames(nestedImageEl: boolean = false): string {
    let classNames;

    // if a boolean value of true is passed this is the nested image element
    if (nestedImageEl) {
        classNames = this.props.round ? 
            `${this.props.managedClass.image} ${this.props.managedClass.image_round}` : this.props.managedClass.image;

       // with this return, nested image instances will exit here
        return classNames;
    }
    // non nested instances run their code here
    classNames = this.props.src ? this.props.managedClasses.image : this.props.managedClasses.picture;

    if (this.props.src && this.props.round) {
        classNames = `${classNames} ${this.props.managedClasses.image_round}`;
    }

    return super.generateClassNames(classNames);
}
```